python/lsst/pipe/drivers/background.py

```diff
@@ -276,15 +276,20 @@ def measureScale(self, image, skyBackground):
         """
         if isinstance(image, afwImage.Exposure):
             image = image.getMaskedImage()
-        xLimits = numpy.linspace(0, image.getWidth() - 1, self.config.xNumSamples + 1, dtype=int)
-        yLimits = numpy.linspace(0, image.getHeight() - 1, self.config.yNumSamples + 1, dtype=int)
+        xNumSamples = min(self.config.xNumSamples, image.getWidth())
```

Comment:
Maybe add a comment here about why you are taking the minimum, in case a future editor of the code is confused